omega/linear_regression/test_linear_regression.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Licensed under the GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.en.html
+
+import unittest
+
+import torch as ch
+import omega as omg
+import torch.utils.data
+
+from omega.test.toolbox import _tempfile, _test_data_path
+
+import logging
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class TestNN(unittest.TestCase):
+    def setUp(self):
+        # sets up a basic input dataset which implements a XOR gate.
+        x = ch.Tensor([24, 50, 15])
+        y = ch.Tensor([21.54945196, 47.46446305, 17.21865634])
+        self.train_data = torch.utils.data.TensorDataset(x, y)
+
+    def model_equal(self, m1, m2):
+        # compares two omg.nn models by comparing their weights and biases
+        for wt1, wt2 in zip(m1.layerwise_weights, m2.layerwise_weights):
+            self.assertTrue(torch.equal(wt1, wt2))
+
+        for b1, b2 in zip(m1.layerwise_biases, m2.layerwise_biases):
+            self.assertTrue(torch.equal(b1, b2))
+
+    def test_consistency(self):
+        # tests for model's load save consistency.
+        old_nw = omg.nn.NetworkMesh([2, 5, 2], seed=100)
```

Comment:
Modify the test_consistency module. I think it is not generic and suitable for all sort of learning technique. I

Reply:
It is not supposed to be generic. It is specific to each model.